Dockerfile

```diff
@@ -1,160 +1,186 @@
-    tar xz "pushgateway-${PUSHGATEWAY_VERSION}.${OS}-${ARCH}/pushgateway" && \
-    mv "pushgateway-${PUSHGATEWAY_VERSION}.${OS}-${ARCH}/pushgateway" /usr/local/bin/pushgateway && \
-    rm -rf "pushgateway-${PUSHGATEWAY_VERSION}.${OS}-${ARCH}" && \
-    chmod +x /usr/local/bin/pushgateway
-
-# Update local copy of AWS IP Ranges.
-RUN curl -fsSL https://ip-ranges.amazonaws.com/ip-ranges.json -o util/ipresolver/aws-ip-ranges.json
-
-RUN ln -s $QUAYCONF /conf && \
-    ln -sf /dev/stdout /var/log/nginx/access.log && \
-    ln -sf /dev/stdout /var/log/nginx/error.log && \
-    chmod -R a+rwx /var/log/nginx
-
-# Cleanup
-RUN UNINSTALL_PKGS="\
-    gcc-c++ git \
-    openldap-devel \
-    gpgme-devel \
-    python3-devel \
-    optipng \
-    kernel-headers \
-    " && \
-    yum remove -y $UNINSTALL_PKGS && \
-    yum clean all && \
-    rm -rf /var/cache/yum /tmp/* /var/tmp/* /root/.cache
+# Ordered from least changing to most changing.
+COPY --from=jwtproxy /usr/local/bin/jwtproxy /usr/local/bin/jwtproxy
+COPY --from=pushgateway /usr/local/bin/pushgateway /usr/local/bin/pushgateway
+COPY --from=build-python /app /app
+COPY --from=config-tool /go/bin/config-tool /bin
+COPY --from=config-editor /build ${QUAYDIR}/config_app
+COPY --from=build-static /build/static ${QUAYDIR}/static
+# Copy in source and update local copy of AWS IP Ranges.
+# This is a bad place to do the curl, but there's no good place to do
+# it except to have it checked in.
+COPY --chown=0:0 . ${QUAYDIR}
+RUN set -ex\
+	; chmod -R g=u "${QUAYDIR}"\
+	; curl -fsSL https://ip-ranges.amazonaws.com/ip-ranges.json -o util/ipresolver/aws-ip-ranges.json\
+	;
 
 EXPOSE 8080 8443 7443 9091 55443
-
-RUN chgrp -R 0 $QUAYDIR && \
-    chmod -R g=u $QUAYDIR
-
-RUN mkdir /datastorage && chgrp 0 /datastorage && chmod g=u /datastorage && \
-    chgrp 0 /var/log/nginx && chmod g=u /var/log/nginx && \
-    mkdir -p /conf/stack && chgrp 0 /conf/stack && chmod g=u /conf/stack && \
-    mkdir -p /tmp && chgrp 0 /tmp && chmod g=u /tmp && \
-    mkdir /certificates && chgrp 0 /certificates && chmod g=u /certificates && \
-    chmod g=u /etc/passwd
-
-
-# Allow TLS certs to be created and installed as non-root user
-RUN chgrp -R 0 /etc/pki/ca-trust/extracted && \
-    chmod -R g=u /etc/pki/ca-trust/extracted && \
-    chgrp -R 0 /etc/pki/ca-trust/source/anchors && \
-    chmod -R g=u /etc/pki/ca-trust/source/anchors && \
-    chgrp -R 0 /usr/local/lib/python$PYTHON_VERSION/site-packages/certifi && \
-    chmod -R g=u /usr/local/lib/python$PYTHON_VERSION/site-packages/certifi
-
-VOLUME ["/var/log", "/datastorage", "/tmp", "/conf/stack"]
-
-USER 1001
-
+# Don't expose /var/log as a volume, because we just configured it
+# correctly above.
+# It's probably unwise to mount /tmp as a volume but if someone must,
+# make sure it's mode 1777 like /tmp should be.
+VOLUME ["/datastorage", "/tmp", "/conf/stack"]
+# In non-Openshift environments, drop privilege.
+USER nobody
```

Comment:
This is preventing the init script from installing the certs in `/etc/pki/ca-trust/source/anchors` due to lack of permission.